testing/test_helpers.py

```diff
@@ -47,6 +50,24 @@ class F(object):
     assert varnames(F) == ((), ())
 
 
+@pytest.mark.skipif(
+    sys.version_info < (3,), reason="Keyword only arguments are Python 3 only"
+)
+def test_varnames_keyword_only():
+    # SyntaxError on Python 2, so we exec
+    global _f1, _f2, _f3
+    exec(
+        "global _f1, _f2, _f3\n"
```

Comment:
@nicoddemus It seems that this is needed only in pytest. What am I not getting?

Reply:
Not sure, might have something to do with assertion rewriting. I would avoid injecting those into the global namespace though:

```python
    ns = {} 
    exec(
        "def _f1(x, *, y): pass\n"
        "def _f2(x, *, y=3): pass\n"
        "def _f3(x=1, *, y=3): pass\n",
        ns
    )
    assert varnames(ns['_f1']) == (("x",), ())
    assert varnames(ns['_f2']) == (("x",), ())
    assert varnames(ns['_f3']) == ((), ("x",))
```

What do you think?

Reply:
Thanks!